easybuild/framework/easyblock.py

```diff
@@ -2892,9 +2894,12 @@ def build_and_install_one(ecdict, init_env):
             buildstats = get_build_stats(app, start_time, build_option('command_line'))
             _log.info("Build stats: %s" % buildstats)
 
-            # for reproducability we dump out the fully processed easyconfig since the contents can be affected
-            # by subtoolchain resolution (and related options) and/or hooks
-            reprod_dir = reproduce_build(app, new_log_dir)
+            # move the reproducability files to the final log directory
+            archive_reprod_dir = os.path.join(new_log_dir, os.path.basename(reprod_dir))
+            if os.path.exists(archive_reprod_dir):
+                print_warning("Reproducability directory %s already exists, cannot overwrite", archive_reprod_dir)
```

Comment:
@ocaisa This will never actually print a warning as is, cfr. https://github.com/ocaisa/easybuild-framework/pull/40 for a fix.

However, I'm a bit worried about this: when `--module-only` is used, this means that the stuff in the `reprod` dir will never get refreshed...

I don't think that's our intention, is it? We *should* be overwriting in that case, no?

It's a tricky one, I guess, since we probably don't want to blindly blow away what is there already either...

Reply:
I was considering doing a bit of inception here and creating a `module_only` subdirectory in the reprod dir for that case and dumping everything there instead. That one would be allowed to be overwritten if the conditions were met again.

Reply:
@ocaisa How about a "safe" variant of `copy_dir` which backs up existing files before overwriting them? That would come in useful here I think, since then you solve the problem of whether or not to replace the existing files under `--module-only`.

Reply:
This doesn't exist yet right? Are you suggesting that I make a back_up directory and move things there? That would be a lot easier than doing it file by file